### PR TITLE
Cherry pick v0.31.1 changes into release-v0.31

### PR DESCRIPTION
**Pull request by rfratto**

### CODE REVIEW

File: docs/sources/upgrade-guide/_index.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ weight: 800
 This guide describes all breaking changes that have happened in prior
 releases and how to migrate to newer versions.
 
-## Unreleased changes
+## v0.31.0
```
Comment (karengermond) [inline docs/sources/upgrade-guide/_index.md:13]: Should this be v0.31.1?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,21 @@ This document contains a historical list of changes between releases. Only
...
 
+v0.31.1 (2023-02-06)
+--------------------
+
+> **BREAKING CHANGES**: This release has breaking changes. Please read entries
+> carefully and consult the [upgrade guide][] for specific instructions.
+
+### Breaking changes
+
+- All release Windows `.exe` files are now published as a zip archive.
```
Comment (ptodev) [inline CHANGELOG.md:18]: Sorry I just realised that it's a bit weird how these changes are listed under 0.31.0 in the upgrade guide, but under 0.31.1 in the changelog?

**Review by ptodev — APPROVED** [review]

**Review by karengermond — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (ptodev) [reply CHANGELOG.md:18]: Maybe we have to list the breaking changes separately for both v0.31.0 and v0.31.1? I.e. for v0.31.0 we have some binaries unzipped, and for v0.31.1 we have all of them zipped.

**Pull request merged**